roomserver/internal/input/input_events.go

```diff
@@ -442,8 +443,11 @@ func (r *Inputer) processRoomEvent(
 		}
 	}
 
-	if err = r.kickGuests(ctx, event, roomInfo); err != nil {
-		logrus.WithError(err).Error("failed to kick guest users on m.room.guest_access revocation")
+	// If guest_access changed and is not can_join, kick all guest users.
+	if event.Type() == gomatrixserverlib.MRoomGuestAccess && gjson.GetBytes(event.Content(), "guest_access").String() != "can_join" {
```

Comment:
Please do not use `.String()` as that type-coerces the value which is not what we want: https://github.com/tidwall/gjson/blob/v1.14.4/gjson.go#L72 - use `.Str` instead which is only populated if the value is in fact a string.